client/dashboard/me/mcp/index.tsx

```diff
@@ -0,0 +1,371 @@
+							) }
+						/>
+
+						{ /* Render tools grouped by category */ }
+						<VStack spacing={ 4 }>
+							{ CATEGORY_ORDER.map( ( categoryName ) => {
+								const categoryTools = groupedTools[ categoryName ];
+								if ( ! categoryTools || categoryTools.length === 0 ) {
+									return null;
+								}
+
+								return (
+									<VStack key={ categoryName } spacing={ 4 }>
+										<Text as="h4" size={ 14 } weight={ 600 }>
+											{ categoryName }
+										</Text>
+										<VStack spacing={ 4 }>
+											{ categoryTools.map( ( [ toolId, tool ]: [ string, McpAbility ] ) => (
+												<ToggleControl
+													key={ toolId }
+													__nextHasNoMarginBottom
+													checked={ tool.enabled }
+													disabled={ mutation.isPending }
+													label={ tool.title }
+													help={ tool.description }
+													onChange={ ( checked ) => handleToolChange( toolId, checked ) }
+												/>
+											) ) }
+										</VStack>
+									</VStack>
+								);
+							} ) }
+						</VStack>
+					</VStack>
+				</CardBody>
+			</Card>
+		);
+	};
+
+	const renderContent = () => {
+		// Use mcpAbilities directly since we're using auto-save
+		const accountToolsToShow = availableTools;
+
+		return (
+			<VStack spacing={ 8 }>
+				{ /* MCP Tool Access Master Toggle */ }
+				<Card>
+					<CardBody>
+						<VStack spacing={ 4 }>
+							<div
+								style={ {
+									display: 'flex',
+									justifyContent: 'space-between',
+									alignItems: 'center',
+								} }
+							>
+								<SectionHeader
+									level={ 3 }
+									title={ __( 'MCP Tool Access' ) }
+									description={ __(
+										'Control which MCP tools can access your WordPress.com account and sites.'
+									) }
+								/>
+								{ anyToolsEnabled && (
+									<Link to="/me/mcp/setup">
+										<Button variant="secondary">{ __( 'Configure MCP Client' ) }</Button>
+									</Link>
+								) }
+							</div>
+
+							<ToggleControl
+								__nextHasNoMarginBottom
+								checked={ anyToolsEnabled }
+								onChange={ handleMasterToggle }
+								label={
+									<Text weight="bold">
+										{ anyToolsEnabled
+											? __( 'Disable MCP Tool Access' )
+											: __( 'Enable MCP Tool Access' ) }
+									</Text>
+								}
+							/>
+						</VStack>
+					</CardBody>
+				</Card>
+
+				{ /* Site-Specific Settings */ }
+				{ hasTools && anyToolsEnabled && (
+					<Card>
+						<CardBody>
+							<VStack spacing={ 4 }>
+								<SectionHeader
+									level={ 3 }
+									title={ __( 'Site-specific MCP settings' ) }
+									description={ __(
+										'Choose a site to block all MCP tools for all users on that site. This overrides your account settings.'
+									) }
+								/>
+
+								<PreferencesLoginSiteDropdown
+									sites={ sites }
+									value={ selectedSiteId }
+									onChange={ ( value ) => setSelectedSiteId( value || '' ) }
```

Comment:
Let's align with other usages to store the value as int, and pass it to this component as string. If so, we don't need to do `parseInt` elsewhere WDTY?